vms/platformvm/txs/builder/builder.go

```diff
@@ -92,6 +96,27 @@ type DecisionTxBuilder interface {
 		ownerAddrs []ids.ShortID,
 		keys []*secp256k1.PrivateKey,
 		changeAddr ids.ShortID,
+		memo []byte,
+	) (*txs.Tx, error)
+
+	NewTransformSubnetTx(
```

Comment:
Added since needed in tests

Reply:
I'd prefer keeping the convention of naming builders as `New + tx_name`.
The unintentional use is an interesting point but I think it's not a problem here. Generally:

- We don't build txs in the nodes usually. X-chain doesn't even have a tx builder package!
- Nodes receives txs either by gossip (no creation there, just parsing and verification) or APIs
- P-chain APIs allowing you to build txs are currently deprecated. Preferred solution is using the wallet package. We may drop the APIs with E upgrade, after which txBuilder will be only a test commodity
- Full discloruse, there is a single tx `RewardValidatorTx` that is generaled automatically by the node when it's time to evict a staker. It has its own builder, whichi we moved outside of txbuilder package recently (it's a simple tx that needs no inputs/outputs, just carries the ID of the staker to be rewarded).